### DISCUSSION TITLE
Support for Apache ECharts

### DISCUSSION THREAD

**Discussion by rohitsathish** [reflex-dev/reflex#2272] [Ideas]
In addition to plotly support, it would be great to have apache echarts. Out of the box, echarts are the most visually appealing and have an incredibly large library of plot types which makes them attractive for visualization-based web apps - https://echarts.apache.org/examples/en/index.html

Competitors like NiceGUI and Solara already have this integration - 
1. https://nicegui.io/documentation/echart
2. https://solara.dev/api/echarts

**Comment by picklelo** [comment]
We're currently using the Recharts library as our core graphing - does that work for you? https://recharts.org/en-US/ (https://reflex.dev/docs/library/graphing/linechart/)

Otherwise, I think this would be great to wrap as a 3rd party library, in the upcoming months we're planning to improve our ecosystem to people can easily wrap / share components from different libraries.

**Comment by snth** [comment]
+1 for ECharts.

I also really like Altair in Python and vega-lite. I was wondering whether with [react-vega](https://github.com/vega/react-vega) it might not be that difficult to vega-lite charts available in Reflex.

I love Reflex btw; thank you so much for creating it! I would just love to add more visualisations and dashboards to my apps.

**Comment by masenf** [comment]
echarts: https://github.com/masenf/reflex-echarts

**Comment by nicedouble** [comment]
I use streamlit to make my app, but I recently found that reflex might suit me better. But unfortunately, reflex does not support echarts.
Waiting official echarts!And need support click event.

**Reply by picklelo** [reply]
@nicedouble see the comment above, we have a third party charts library now: https://github.com/masenf/reflex-echarts

**Reply by Aneeqasif** [reply]
> @nicedouble see the comment above, we have a third party charts library now: https://github.com/masenf/reflex-echarts

this seems to be stale project and not well integrated any update on this , echarts is a very very powerfull library widely adapted with over 60k stars on github, if that counts , it would be nice to use it in our fav python framework for developing custom dashboards. Thanks

**Comment by remsidave** [comment]
My vote goes to vega and vega-lite (or altair in python). They are super powerful and easy to learn. Streamlit for instance already supports them and we use st.vega_lite_chart() very often
